### PR TITLE
stored: use "archive_device_string" variable for archive device

### PR DESCRIPTION
**Pull request by pstorz**
The "archive device" directive was stored in the variable "device_name"
in the "DeviceResource" Class and in "dev_name" in the "Device" Class.

This is misleading as the device name (i.e. the name of the device
resource) is stored in the "resource_name_" member of the device resource.

#### Consider the following chapters in the Bareos Developer Documentation

- [x] [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [x] [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [x] [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

#### Add some information

- [x] Add a small description to the CHANGELOG.md file and refer to your PR using this syntax '[PR #xyz]'
- [x] Add your name to the AUTHORS file

#### Keep spirit!
- [x] Do not be afraid to hand in a PR!

### CODE REVIEW

File: core/src/plugins/stored/scsicrypto/scsicrypto-sd.cc [hunk 1]
```diff
@@ -340,7 +340,7 @@ static bRC do_set_scsi_encryption_key(void* value)
      * Check if encryption key is needed for reading this volume.
      */
     P(crypto_operation_mutex);
-    if (!NeedScsiCryptoKey(dev->fd(), dev->dev_name, true)) {
+    if (!NeedScsiCryptoKey(dev->fd(), dev->archive_device, true)) {
```
Comment (franku) [inline core/src/plugins/stored/scsicrypto/scsicrypto-sd.cc:343]:
```suggestion
    if (!NeedScsiCryptoKey(dev->fd(), dev->archive_device_name, true)) {
```

**Review by franku — APPROVED** [review]
Good idea to give a meaningful name.. 

However, please update the changelog.

File: core/src/plugins/stored/scsicrypto/scsicrypto-sd.cc (hunk 1 above)
Comment (pstorz) [reply core/src/plugins/stored/scsicrypto/scsicrypto-sd.cc:343]: After a second thought I think that "archive_device" is the right name, as the directive that sets the value is "archive device",
See stored_conf.cc:

{"ArchiveDevice", CFG_TYPE_STRNAME, ITEM(res_dev, device_name), 0, CFG_ITEM_REQUIRED, NULL, NULL, NULL},

Comment (franku) [reply core/src/plugins/stored/scsicrypto/scsicrypto-sd.cc:343]: Suggestion: archive_device_string

**Pull request merged**